roofit/roofit/src/RooLagrangianMorphFunc.cxx

```diff
@@ -0,0 +1,3511 @@
+  RooRealVar *nNP0 = new RooRealVar("nNP0", "nNP0", 1., 0, 1.);
+  nNP0->setStringAttribute("NP", "0");
+  nNP0->setConstant(true);
+  this->_flags.add(*nNP0);
+  RooRealVar *nNP1 = new RooRealVar("nNP1", "nNP1", 1., 0, 1.);
+  nNP1->setStringAttribute("NP", "1");
+  nNP1->setConstant(true);
+  this->_flags.add(*nNP1);
+  RooRealVar *nNP2 = new RooRealVar("nNP2", "nNP2", 1., 0, 1.);
+  nNP2->setStringAttribute("NP", "2");
+  nNP2->setConstant(true);
+  this->_flags.add(*nNP2);
+  RooRealVar *nNP3 = new RooRealVar("nNP3", "nNP3", 1., 0, 1.);
+  nNP3->setStringAttribute("NP", "3");
+  nNP3->setConstant(true);
+  this->_flags.add(*nNP3);
+  RooRealVar *nNP4 = new RooRealVar("nNP4", "nNP4", 1., 0, 1.);
+  nNP4->setStringAttribute("NP", "4");
+  nNP4->setConstant(true);
+  this->_flags.add(*nNP4);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// copy constructor
+
+RooLagrangianMorphFunc::RooLagrangianMorphFunc(
+    const RooLagrangianMorphFunc &other, const char *name)
+    : RooAbsReal(other, name), _cacheMgr(other._cacheMgr, this),
+      _scale(other._scale), _sampleMap(other._sampleMap),
+      _physics(other._physics.GetName(), this, other._physics),
+      _operators(other._operators.GetName(), this, other._operators),
+      _observables(other._observables.GetName(), this, other._observables),
+      _binWidths(other._binWidths.GetName(), this, other._binWidths),
+      _flags(other._flags.GetName(), this, other._flags),
+      _config(other._config), _curNormSet(0) {
+  std::cout << "COPY CONSTRUCTOR" << std::endl;
+  for (size_t j = 0; j < other._diagrams.size(); ++j) {
+    std::vector<RooListProxy *> diagram;
+    for (size_t i = 0; i < other._diagrams[j].size(); ++i) {
+      RooListProxy *list = new RooListProxy(other._diagrams[j][i]->GetName(),
+                                            this, *(other._diagrams[j][i]));
+      diagram.push_back(list);
+    }
+    this->_diagrams.push_back(diagram);
+  }
+  TRACE_CREATE
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// set energy scale of the EFT expansion
+
+void RooLagrangianMorphFunc::setScale(double val) { this->_scale = val; }
+
+////////////////////////////////////////////////////////////////////////////////
+/// get energy scale of the EFT expansion
+
+double RooLagrangianMorphFunc::getScale() { return this->_scale; }
+
+////////////////////////////////////////////////////////////////////////////////
+// default constructor
+
+RooLagrangianMorphFunc::RooLagrangianMorphFunc()
+    : _operators("operators", "set of operators", this, kTRUE, kFALSE),
+      _observables("observable", "morphing observable", this, kTRUE, kFALSE),
+      _binWidths("binWidths", "set of bin width objects", this, kTRUE, kFALSE) {
+  static int counter(0);
+  //  DEBUG("default constructor called: " << this << " " << counter);
+  counter++;
+  this->printAuthors();
```

Comment:
```suggestion
```
I don't think printing the authors every time a `RooLagrangianMorphFunc` gets created is necessary.